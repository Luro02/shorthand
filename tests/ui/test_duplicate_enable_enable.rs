use shorthand::ShortHand;

#[derive(ShortHand)]
#[shorthand(enable(copy))]
struct Example {
    // enabling an attribute twice
    #[shorthand(enable(const_fn, const_fn))]
    value_1: usize,
    #[shorthand(enable(const_fn))]
    #[shorthand(enable(const_fn))]
    value_2: usize,
    // enabling an attribute, that is already enabled globally
    #[shorthand(enable(copy))]
    value_3: usize,
}

fn main() {}
