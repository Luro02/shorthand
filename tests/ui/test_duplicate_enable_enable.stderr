error: duplicate field `const_fn`
 --> $DIR/test_duplicate_enable_enable.rs:7:34
  |
7 |     #[shorthand(enable(const_fn, const_fn))]
  |                                  ^^^^^^^^

error: redundant field `const_fn`, which is already enabled
  --> $DIR/test_duplicate_enable_enable.rs:10:24
   |
10 |     #[shorthand(enable(const_fn))]
   |                        ^^^^^^^^

error: redundant field `copy`, which is already enabled
  --> $DIR/test_duplicate_enable_enable.rs:13:24
   |
13 |     #[shorthand(enable(copy))]
   |                        ^^^^
